File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/command/ProjectCommand.java
package dev.jefy.connectpro.portfolio.applicaion.command;

import org.jspecify.annotations.NullMarked;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

import dev.jefy.connectpro.portfolio.applicaion.dtos.ProjectRequest;
import dev.jefy.connectpro.shared.domain.vo.ImageUrl;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface ProjectCommand {
    ProjectId create(ProjectRequest request);
    ProjectId update(ProjectId projectId, ProjectRequest request);
    ProjectId addImage(ProjectId projectId, MultipartFile image) throws IOException;
    ProjectId removeImage(ProjectId projectId, ImageUrl imageUrl) throws IOException;
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/command/ProjectCommand.kt
package dev.jefy.connectpro.portfolio.application.command

import dev.jefy.connectpro.portfolio.application.dtos.ProjectRequest
import dev.jefy.connectpro.portfolio.domain.vo.ProjectId
import dev.jefy.connectpro.shared.application.dtos.ImageRequest
import org.springframework.web.multipart.MultipartFile
import java.io.IOException

interface ProjectCommand {
    fun create(request: ProjectRequest): ProjectId
    fun update(projectId: ProjectId, request: ProjectRequest): ProjectId
    @Throws(IOException::class)
    fun addImage(projectId: ProjectId, image: MultipartFile): ProjectId
    @Throws(IOException::class)
    fun removeImage(projectId: ProjectId, image: ImageRequest): ProjectId
    @Throws(IOException::class)
    fun delete(projectId: ProjectId)
}